dlc-manager/src/sub_channel_manager.rs

```diff
@@ -692,80 +697,218 @@ where
+                                index: 0,
+                            },
+                            c.ln_rollback.channel_value_satoshis,
+                            c.ln_rollback.value_to_self_msat,
+                        );
+                        Ok(())
+                    },
+                )?;
+                sub_channel.state = SubChannelState::Signed(c.signed_subchannel);
+                self.force_close_signed_channel(sub_channel, Some(c.commitment_transactions))?;
+            }
+            SubChannelState::CloseConfirmed(_) => {
+                self.force_close_offered_channel(sub_channel)?;
+            }
+            SubChannelState::OnChainClosed
+            | SubChannelState::CounterOnChainClosed
+            | SubChannelState::OffChainClosed
+            | SubChannelState::ClosedPunished(_)
+            | SubChannelState::Rejected
+            | SubChannelState::Closing(_) => {
+                error!(
+                    "Tried to force close channel with {:?} state",
+                    sub_channel.state
+                );
+                return Err(Error::InvalidParameters(
+                    "Channel is not in a state to be force closed".to_string(),
+                ));
+            }
+        };
+
+        Ok(())
+    }
+
+    fn force_close_offered_channel(&self, mut sub_channel: SubChannel) -> Result<(), Error> {
+        let (closed_channel, closed_contract) = self.get_closed_dlc_channel_and_contract(
+            sub_channel
+                .get_dlc_channel_id(0)
+                .expect("to have a channel id in offered state"),
+            false,
         )?;
-        let counter_party = signed.counter_party;
-        self.ln_channel_manager.with_channel_lock_no_check(
-            channel_id,
-            &counter_party,
-            |channel_lock| {
-                let publish_base_secret = self
-                    .dlc_channel_manager
-                    .get_wallet()
-                    .get_secret_key_for_pubkey(&signed.own_base_points.publish_basepoint)?;
+        let dlc_channel_id = sub_channel.get_dlc_channel_id(0);
+        sub_channel.state = SubChannelState::OnChainClosed;
+        self.ln_channel_manager
+            .force_close_channel(&sub_channel.channel_id, &sub_channel.counter_party)?;
+        self.dlc_channel_manager
+            .get_store()
+            .upsert_sub_channel(&sub_channel)?;
+        self.dlc_channel_manager
+            .get_store()
+            .upsert_channel(closed_channel, Some(closed_contract))?;
+        //TODO(tibo): this is actually unsafe, we shouldn't clean up the chain monitor before
+        //having the commitment transaction confirmed on chain.
+        let mut chain_monitor = self.dlc_channel_manager.get_chain_monitor().lock().unwrap();
+        chain_monitor.cleanup_channel(sub_channel.channel_id);
+        if let Some(dlc_channel_id) = dlc_channel_id {
+            chain_monitor.cleanup_channel(dlc_channel_id);
+        }
```

Comment:
We can not cleanup, but it means storage will just keep growing as channels get open/closed. It's probably not a big issue and I'm fine either way. But if we do want to cleanup, it's required here because our chain monitor will not react to commitment transactions.